cohttp-eio/src/client.ml

```diff
@@ -62,16 +70,21 @@ let response buf_read =
 
 (* Generic HTTP call *)
 
-let call ?(meth = `GET) ?(version = `HTTP_1_1) ?(headers = Http.Header.init ())
-    ?(body = Body.Empty) ~conn host resource_path =
-  let host =
-    match host with
-    | host, Some port -> host ^ ":" ^ string_of_int port
-    | host, None -> host
+let call ?meth ?version ?(headers = Http.Header.init ()) ?(body = Body.Empty)
+    ~conn host resource_path =
+  let headers =
+    if not (Http.Header.mem headers "Host") then
+      let host =
+        match host with
+        | host, Some port -> host ^ ":" ^ string_of_int port
+        | host, None -> host
+      in
+      Http.Header.add_unless_exists headers "Host" host
```

Comment:
```suggestion
      Http.Header.add "Host" host
```
If you introduce the check in line 76, there is no need to use the extra logic. Why did you decide to change it? It is morally equivalent to the one before

Reply:
Indeed, good catch. Fixed now.